Carbon.Security/Functions/Grant-CPrivilege.ps1

```diff
@@ -0,0 +1,162 @@
+     * SeShutdownPrivilege
+     * SeSyncAgentPrivilege
+     * SeSystemEnvironmentPrivilege
+     * SeSystemProfilePrivilege
+     * SeSystemtimePrivilege
+     * SeTakeOwnershipPrivilege
+     * SeTcbPrivilege
+     * SeTimeZonePrivilege
+     * SeTrustedCredManAccessPrivilege
+     * SeUndockPrivilege
+     * SeUnsolicitedInputPrivilege
+
+    [Account Right Constants](https://learn.microsoft.com/en-us/windows/win32/secauthz/account-rights-constants):
+
+     * SeBatchLogonRight
+     * SeDenyBatchLogonRight
+     * SeDenyInteractiveLogonRight
+     * SeDenyNetworkLogonRight
+     * SeDenyRemoteInteractiveLogonRight
+     * SeDenyServiceLogonRight
+     * SeInteractiveLogonRight
+     * SeNetworkLogonRight
+     * SeServiceLogonRight
+
+    .LINK
+    Get-CPrivilege
+
+    .LINK
+    Revoke-CPrivilege
+
+    .LINK
+    Test-CPrivilege
+
+    .LINK
+    Test-CPrivilegeName
+
+    .LINK
+    https://learn.microsoft.com/en-us/windows/win32/secauthz/privilege-constants
+
+    .LINK
+    https://learn.microsoft.com/en-us/windows/win32/secauthz/account-rights-constants
+
+    .EXAMPLE
+    Grant-CPrivilege -Identity Batcomputer -Privilege SeServiceLogonRight
+
+    Grants the Batcomputer account the ability to logon as a service.
+    #>
+    [CmdletBinding()]
+    param(
+        # The user/group name to grant rights/privileges.
+        [Parameter(Mandatory)]
+        [String] $Identity,
+
+        # The rights/privileges to grant.
+        #
+        # [Privilege names are documented on the "Privilege Constants"
+        # page.](https://learn.microsoft.com/en-us/windows/win32/secauthz/privilege-constants)
+        #
+        # [Rights names are documented on the "Account Rights Constants"
+        # page.](https://learn.microsoft.com/en-us/windows/win32/secauthz/account-rights-constants)
+        [Parameter(Mandatory)]
+        [String[]] $Privilege
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+    $account = Resolve-CIdentity -Name $Identity
+    if( -not $account )
+    {
+        return
+    }
+
+    $privilegesToGrant = $Privilege | Where-Object { -not (Test-CPrivilege -Identity $account.FullName -Privilege $_) }
+    if (-not $privilegesToGrant)
+    {
+        return
+    }
+
+    $unknownPrivileges = $privilegesToGrant | Where-Object { -not (Test-CPrivilegeName -Name $_) }
+    if ($unknownPrivileges)
+    {
+        $privileges = 'privilege'
+        $thatThose = 'that'
+        $isAre = 'is'
+        if (($unknownPrivileges | Measure-Object).Count -gt 1)
+        {
+            $privileges = 'privileges'
+            $thatThose = 'those'
+            $isAre = 'are'
+        }
+        $msg = "Failed to grant the $($account.FullName) account $($unknownPrivileges -join ', ') ${privileges} " +
+               "because ${thatThose} ${privileges} ${isAre} unknown."
+        Write-Error -Message $msg -ErrorAction $ErrorActionPreference

```

Comment:
Shouldn't the function `return` after this point?

Reply:
No, because the user can pass in a list of privileges and some of them might be known/acceptable. Do as much as possible.